app/gui/view/component-browser/searcher-list-panel/src/lib.rs

```diff
@@ -0,0 +1,662 @@
+        let display_object = display::object::Instance::new(&logger);
+
+        let background = background::View::new(&logger);
+        display_object.add_child(&background);
+        app.display.default_scene.layers.below_main.add_exclusive(&background);
+
+        let favourites_section = Self::init_column_section(&app);
+        let local_scope_section = Self::init_wide_section(&app);
+        let sub_modules_section = Self::init_column_section(&app);
+
+        let scroll_area = ScrollArea::new(&app);
+        display_object.add_child(&scroll_area);
+
+        let camera = &scroll_area.content_layer().camera();
+        let parent_layer = scroll_area.content_layer();
+        let layers = Layers::new(&app.logger, camera, parent_layer);
+
+        favourites_section.set_parent(scroll_area.content());
+        local_scope_section.set_parent(scroll_area.content());
+        sub_modules_section.set_parent(scroll_area.content());
+
+        // Required for correct clipping. The components need to be set up with the
+        // `scroll_area.content_layer` to be masked correctly by the [`ScrollArea`].
+        favourites_section.set_layers(&layers, scroll_area.content_layer());
+        local_scope_section.set_layers(&layers, scroll_area.content_layer());
+        sub_modules_section.set_layers(&layers, scroll_area.content_layer());
+
+        Self {
+            app,
+            display_object,
+            background,
+            scroll_area,
+            favourites_section,
+            local_scope_section,
+            sub_modules_section,
+            layers,
+            logger,
+        }
+    }
+
+    fn init_column_section(app: &Application) -> ColumnSection {
+        let content = app.new_view::<column_grid::ColumnGrid>();
+        LabeledSection::new(content, app)
+    }
+
+    fn init_wide_section(app: &Application) -> WideSection {
+        let content = app.new_view::<component_group::wide::View>();
+        content.set_no_items_label_text("No Entries.");
+        LabeledSection::new(content, app)
+    }
+
+    fn update_layers(&self) {
+        self.favourites_section.set_layers(&self.layers, self.scroll_area.content_layer());
+        self.local_scope_section.set_layers(&self.layers, self.scroll_area.content_layer());
+        self.sub_modules_section.set_layers(&self.layers, self.scroll_area.content_layer());
+    }
+
+    fn update_style(&self, style: &Style) {
+        self.sub_modules_section.set_style(style);
+        self.local_scope_section.set_style(style);
+        self.favourites_section.set_style(style);
+
+        self.background.bg_color.set(style.content.background_color.into());
+        self.background.size.set(style.size());
+
+        self.local_scope_section
+            .content
+            .set_position_x(style.content.padding + style.content.size.x / 2.0);
+        self.local_scope_section
+            .content
+            .set_width(style.size_inner().x - 2.0 * style.content.padding);
```

Comment:
If we refactor `self.local_scope_section.content` to vars, then these 6 lines will turn into 3 lines